### ISSUE TITLE
Upgrade React Native Clipboard to React Native Community's Clipboard before depreciation

### ISSUE THREAD

**Issue by jd-alexander** [wordpress-mobile/gutenberg-mobile#4219]
**Describe the bug**

The `Clipboard` module within `react-native` will be removed in a future release so it will need upgrading. Since [react-native-clipboard/clipboard](https://github.com/react-native-clipboard/clipboard) is readily available for the transition the process should be straight forward. 
```
 console.warn
    Clipboard has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from `@react-native-community/clipboard` instead of 'react-native'. See https://github.com/react-native-clipboard/clipboard

      206 | 
      207 | 	async function getURLFromClipboard() {
    > 208 | 		const clipboardText = await Clipboard.getString();
          | 		                            ^
      209 | 
      210 | 		if ( ! clipboardText ) {
      211 | 			return;
```
**Next steps** 
The effort would involve:
1. Replace all usages of the `Clipboard` from `react-native`. 
2. Fixing all tests that mock the `Clipboard` to utilize the [mocking capabilities](https://github.com/react-native-clipboard/clipboard#mocking-clipboard) built into `@react-native-community/clipboard`

**To Reproduce**
1. Run the native test suite via  `npm run native test > log.txt` 
2. Notice the aforementioned warning in the console.

**Expected behavior**
No warning should be in the console.

**Comment by dcalhoun** [comment]
FYI I committed https://github.com/WordPress/gutenberg/pull/36328/commits/d6cf5d735593c7c1f9209912d7d315e78e0dc81b, which resolves this in the React Native 0.66 upgrade. So, we could await PR merge or cherry pick the commit, if desired.

**Comment by ttahmouch** [comment]
I forgot to reassign the issue per our Asana conversation yesterday, @dcalhoun @antonis .
